.github/workflows/docker-image.yaml

```
@@ -0,0 +1,33 @@
+name: Build and Push Docker Image
+
+on:
+  push:
+    branches:
+      - master
```

```suggestion
      - main
```

Aren't our default branch is main?

---

Beyond this, I think that it makes sense that the trigger for this is a release and not a push to `main` as we would only want full releases to go to Docker Hub.